### ISSUE TITLE
Rename QGIS_PLUGINPATH to QGIS_PYTHON_PLUGIN_PATH

### ISSUE THREAD

**Issue by elpaso** [qgis/qgis3.0_api#78]
Deprecate env var `QGIS_PLUGINPATH` in favour of `QGIS_PYTHON_PLUGINS_PATH`

How is the deprecation policy?
I'd like to support the old `QGIS_PLUGINPATH` but raise a deprecation warning in the logs.

Side note : do we have a comprehensive list of all the env vars used in QGIS?

**Comment by NathanW2** [comment]
Why even remove it?  Does it matter a lot what it's called?

**Comment by elpaso** [comment]
@NathanW2  motivation is consistency and meaning:

1. Other vars contain and underscore before `PATH`
2. there are also C++ plugins, auth plugins etc., this env var is specific to Python plugins, to avoid confusion it should be renamed.

I always thought I made a mistake choosing that name.

**Comment by NathanW2** [comment]
IMO just add a new environment variable that is a alias.  So we can support both.  I guess you can add a deprecation warning but I suspect it would be ignored really.

**Comment by NathanW2** [comment]
Oh right reading it again that is what you want to do.

**Comment by elpaso** [comment]
Yes, as I said I wanted to support both in code, but the new one should be the one in the QGIS 3.0 docs.

**Comment by m-kuhn** [comment]
I was also always a bit surprised by the name of this variable and the way you propose to do it is a straightforward and low-risk migration path. Sounds good to me.
